core/src/main/java/org/elasticsearch/ingest/core/Pipeline.java

```diff
@@ -83,9 +85,10 @@ public String getDescription() {
     public final static class Factory {
 
         public Pipeline create(String id, Map<String, Object> config, Map<String, Processor.Factory> processorRegistry) throws Exception {
+            AtomicInteger processorIdGenerator = new AtomicInteger(0);
```

Comment:
given how processorIdGenerator is used, it really shouldn't be AtomicInteger anymore, otherwise it is misleading, a simple int counter does it instead.